be/src/olap/merger.cpp

```diff
@@ -41,6 +41,10 @@ Status Merger::merge_rowsets(TabletSharedPtr tablet, ReaderType reader_type,
     reader_params.reader_type = reader_type;
     reader_params.rs_readers = src_rowset_readers;
     reader_params.version = dst_rowset_writer->version();
+    reader_params.delete_predicates = tablet->delete_predicates();
+    std::copy(tablet->delete_predicates().cbegin(), tablet->delete_predicates().cend(),
+              std::inserter(reader_params.delete_predicates,
+                            reader_params.delete_predicates.begin()));
```

Comment:
Why not acquire tablet lock during init reader_params's delete predicate property?

Reply:
1. compaction don't need to care new added delete predicates, it's useless for this compaction.
2. only compaction would remove delete predicates, so it also don't need to worry other thread to remove the delete predicates related with input rowset.

Reply:
I added the lock finally, it's more easier to understand and cost very low.